libcontainer/process_linux.go

```diff
@@ -141,6 +141,16 @@ func (p *setnsProcess) execSetns() error {
 		p.cmd.Wait()
 		return newSystemErrorWithCause(err, "reading pid from init pipe")
 	}
+
+	// Clean up the zombie parent process
+	firstChildProcess, err := os.FindProcess(pid.PidFirstChild)
+	if err != nil {
```

Comment:
In which case the `Wait` is what will fail if the process doesn't exist -- so maybe some thing more like

```go
if _, err := firstChildProcess.Wait(); err != nil {
	if err != unix.ECHILD && err != unix.ESRCH {
		return err
	}
}
```

Maybe?